### PR TITLE
feat: Change proxy shared secret support for TACACS+ proxy clients to be based on loopback or cli config not upstream

### PR DESCRIPTION
**Pull request by Rod-Persky**
This pull request adds support for configuring the expected obfuscation policy for downstream TACACS+ proxy traffic, based on either the highest-priority filtered local proxy endpoint or a new `--proxy-shared-secret` CLI flag. It also refactors the configuration filter logic to propagate this policy and updates related tests accordingly.

**Configuration and Obfuscation Policy Handling:**

* Introduced a new CLI flag `--proxy-shared-secret` in `cli.rs`, allowing operators to specify a shared secret for raw TACACS+ proxy clients when not in SONiC mode. This flag is required if `--proxy-endpoint` is set and conflicts with `--sonic`.
* Updated the configuration filter logic in `config_filter.rs` to:
  * Track and propagate the expected downstream obfuscation policy (`ProxyDownstreamObfuscation`) alongside the filtered TACACS+ configuration, using a new `FilteredTacacsPlus` struct.
  * When local proxy endpoint rows are filtered, the policy is derived from the highest-priority matching row's shared secret, or falls back to the CLI flag if present. [[1]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aR135-R157) [[2]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aR167-R188)
  * Updated the filter constructors and trait implementations to support the new policy propagation. [[1]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aR3-R121) [[2]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aR167-R188)

**Documentation:**

* Expanded the `README.md` to clarify how the proxy obfuscation policy is selected, how the new CLI flag interacts with configuration, and what is expected from local clients under various scenarios.

**Testing:**

* Refactored and extended tests in `config_filter.rs` to verify that the correct obfuscation policy is applied and propagated for various configurations, including edge cases where no shared secret is present. [[1]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aL131-R274) [[2]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aL164-R293) [[3]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aL181-R314) [[4]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aL199-R350) [[5]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aL224-R375) [[6]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aL251-R392) [[7]](diffhunk://#diff-ca04353a9bef62b2a3f6bf6627077ab76f8a7eed67192cfa2d2045586283bb1aL263-R404)

### CODE REVIEW

File: libraries/tacacsrs_sonic/src/mapping.rs [hunk 1]
```diff
@@ -595,7 +595,7 @@ mod tests {
         .expect("previous mapping succeeds");
 
         let mut new_servers = BTreeMap::new();
-        new_servers.insert("192.0.2.10".to_string(), h(&[("priority", "1"), ("passkey", "x")]));
+        new_servers.insert("192.0.2.10".to_string(), h(&[("priority", "10"), ("passkey", "x")]));
```
[the remark this answers is not in the document]
Comment (Rod-Persky) [reply libraries/tacacsrs_sonic/src/mapping.rs:606]: Incorrect, actually higher is preferred. The docs in this repo isn't correct.

File: libraries/tacacsrs_sonic/src/mapping.rs [hunk 2]
```diff
@@ -580,7 +580,7 @@ mod tests {
             .iter()
             .map(|s| s.address.clone())
             .collect::<Vec<_>>();
-        assert_eq!(order, vec!["192.0.2.10", "192.0.2.30", "192.0.2.20"]);
+        assert_eq!(order, vec!["192.0.2.20", "192.0.2.10", "192.0.2.30"]);
```
[the remark this answers is not in the document]
Comment (Rod-Persky) [reply libraries/tacacsrs_sonic/src/mapping.rs:591]: Incorrect, actually higher is preferred. The docs in this repo isn't correct.

File: libraries/tacacsrs_sonic/src/mapping.rs [hunk 3]
```diff
@@ -430,15 +430,15 @@ mod tests {
...
         let cfg = map_sonic_tables_to_tacacs_plus(&tables()).expect("mapping succeeds");
         assert_eq!(cfg.server.len(), 2);
-        // Lower priority -> higher preference (index 0).
-        assert_eq!(cfg.server[0].address, "192.0.2.10");
+        // Higher priority -> higher preference (index 0).
+        assert_eq!(cfg.server[0].address, "192.0.2.20");
         assert_eq!(cfg.server[0].port, 49);
         assert_eq!(cfg.server[0].timeout, 7);
-        assert_eq!(cfg.server[0].shared_secret.as_deref(), Some("default-secret"));
+        assert_eq!(cfg.server[0].shared_secret.as_deref(), Some("per-server-secret"));
```
[the remark this answers is not in the document]
Comment (Rod-Persky) [reply libraries/tacacsrs_sonic/src/mapping.rs:446]: Incorrect, actually higher is preferred. The docs in this repo isn't correct.

File: libraries/tacacsrs_sonic/src/mapping.rs [hunk 4]
```diff
@@ -79,7 +79,7 @@ impl SonicTacacsTables {
 /// Translate a SONiC ConfigDB snapshot into the YANG `TacacsPlus` root.
 ///
 /// Each `TACPLUS_SERVER|<addr>` row becomes one
-/// [`tacacsrs_config::TacacsPlusServer`]. Servers are ordered by ascending
+/// [`tacacsrs_config::TacacsPlusServer`]. Servers are ordered by descending
```
[the remark this answers is not in the document]
Comment (Rod-Persky) [reply libraries/tacacsrs_sonic/src/mapping.rs:82]: Incorrect, actually higher is preferred. The docs in this repo isn't correct.

File: libraries/tacacsrs_sonic/src/mapping.rs [hunk 5]
```diff
@@ -201,7 +201,7 @@ impl SonicServerRow {
     fn from_hash(address: &str, hash: &SonicHash) -> anyhow::Result<Self> {
         let mut row = Self {
             address: address.to_string(),
-            priority: i64::MAX,
+            priority: i64::MIN,
```
[the remark this answers is not in the document]
Comment (Rod-Persky) [reply libraries/tacacsrs_sonic/src/mapping.rs]: Incorrect, actually higher is preferred. The docs in this repo isn't correct.

File: libraries/tacacsrs_sonic/src/mapping.rs [hunk 6]
```diff
@@ -112,11 +112,11 @@ pub fn map_sonic_tables_to_tacacs_plus(tables: &SonicTacacsTables) -> anyhow::Re
...
-    // SONiC convention: lower priority number = higher preference. Ties broken
+    // SONiC convention: higher priority number = higher preference. Ties broken
     // by lexicographic address so the order is deterministic in tests/logs.
     rows.sort_by(|a, b| {
-        a.priority
-            .cmp(&b.priority)
+        b.priority
+            .cmp(&a.priority)
             .then_with(|| a.address.cmp(&b.address))
     });
```
[the remark this answers is not in the document]
Comment (Rod-Persky) [reply libraries/tacacsrs_sonic/src/mapping.rs]: Incorrect, actually higher is preferred. The docs in this repo isn't correct.
